### PR TITLE
SearchableEntityParams

### PR DESCRIPTION
**Pull request by agzam**
> **Epic** #
> Close https://github.com/threatgrid/iroh/issues/5269
> Related #

Adds SearchabelEntityParams

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

bffa633c * origin/searchable-entity-fields_#iroh/5269 remove :query key from entities
51bed4c4 * 1126 adds :query to SearchableEntityParams
6d8e5d31 * adds s/defschema SearchableEntityParams

### CODE REVIEW

File: src/ctia/http/routes/crud.clj [hunk 1]
```diff
@@ -164,13 +164,9 @@
...
          :as services} :- APIHandlerServices]
   (let [capitalized (capitalize-entity entity)
-        ;; Adding additional query params for ES Fulltext search
         search-q-params* (st/merge
                           search-q-params
-                          {(s/optional-key :query_mode)
-                           (describe FullTextQueryMode "Elasticsearch Fulltext Query Mode. Defaults to query_string")
-
-                           ;; We cannot name the parameter :fields, because we already have :fields (part
+                          {;; We cannot name the parameter :fields, because we already have :fields (part
```
Comment (agzam) [inline src/ctia/http/routes/crud.clj:169]: I leave this here for until I get the ES :fields parameter right (in a separate PR)

**Review by ereteog — COMMENTED** [review]
You need to add `query`, which is duplicated in all entities to `SearchableEntityParams`.

**Comment by ereteog** [comment]
Now that you added it to the common schema, you can remove it in each entity.

**Review by ereteog — APPROVED** [review]
LGTM

**Pull request merged**